[Bug]: Execution failed for task ':unityLibrary:compileReleaseJavaWithJavac'

### What happened?

I have installed the latest OneSignal version 3.0.8 from Scoped Registries. After installation I ran OneSignal SDK Setup without errors.

When I then try to build for my Android device I get this error:
CommandInvokationFailure: Gradle build failed. 

* What went wrong:
Execution failed for task ':unityLibrary:compileReleaseJavaWithJavac'.

Not sure how to solve this :-/


### Steps to reproduce?

```Markdown
1. Install OneSignal from Scoped Registries
2. build fro Android
```


### What did you expect to happen?

The app should build and everything should be fine.

### Unity version

2021.3.16f1 (latest)

### OneSignal Unity SDK version

3.0.8

### Platform

Android

### Relevant log output

```Shell
CommandInvokationFailure: Gradle build failed. 
/Applications/Unity/Hub/Editor/2021.3.16f1/PlaybackEngines/AndroidPlayer/OpenJDK/bin/java -classpath "/Applications/Unity/Hub/Editor/2021.3.16f1/PlaybackEngines/AndroidPlayer/Tools/gradle/lib/gradle-launcher-6.1.1.jar" org.gradle.launcher.GradleMain "-Dorg.gradle.jvmargs=-Xmx4096m" "assembleRelease"

Environment Variables:
BURST_ANDROID_MIN_API_LEVEL = 22
XPC_FLAGS = 0x0
JAVA_TOOL_OPTIONS = -Dfile.encoding=UTF-8
MallocNanoZone = 0
ORIGINAL_XDG_CURRENT_DESKTOP = undefined
SHELL = /bin/zsh
SSH_AUTH_SOCK = /private/tmp/com.apple.launchd.z0sW2UrI0O/Listeners
HOME = /Users/jespermansa
PATH = /usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin:/Library/Frameworks/Mono.framework/Versions/Current/Commands

__CFBundleIdentifier = com.unity3d.unityhub
USER = jespermansa
__CF_USER_TEXT_ENCODING = 0x1F5:0x0:0x9
TMPDIR = /var/folders/kr/2715rkvj20q1pp3bxk02klfw0000gn/T/
LOGNAME = jespermansa
JAVA_HOME = /Applications/Unity/Hub/Editor/2021.3.16f1/PlaybackEngines/AndroidPlayer/OpenJDK
ANDROID_NDK_ROOT = /Applications/Unity/Hub/Editor/2021.3.16f1/PlaybackEngines/AndroidPlayer/NDK
XPC_SERVICE_NAME = application.com.unity3d.unityhub.32073928.32073935
COMMAND_MODE = unix2003


stderr[
Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF-8
/Users/jespermansa/Numfeud2023b/Library/Bee/Android/Prj/IL2CPP/Gradle/unityLibrary/src/main/java/com/onesignal/UnityIAMLifecycleHandler.java:30: error: cannot find symbol
public final class UnityIAMLifecycleHandler extends OSInAppMessageLifecycleHandler {
                                                    ^
  symbol: class OSInAppMessageLifecycleHandler
/Users/jespermansa/Numfeud2023b/Library/Bee/Android/Prj/IL2CPP/Gradle/unityLibrary/src/main/java/com/onesignal/UnityIAMLifecycleHandler.java:45: error: cannot find symbol
    public void onWillDisplayInAppMessage(OSInAppMessage message) {
                                          ^
  symbol:   class OSInAppMessage
  location: class UnityIAMLifecycleHandler
/Users/jespermansa/Numfeud2023b/Library/Bee/Android/Prj/IL2CPP/Gradle/unityLibrary/src/main/java/com/onesignal/UnityIAMLifecycleHandler.java:51: error: cannot find symbol
    public void onDidDisplayInAppMessage(OSInAppMessage message) {
                                         ^
  symbol:   class OSInAppMessage
  location: class UnityIAMLifecycleHandler
/Users/jespermansa/Numfeud2023b/Library/Bee/Android/Prj/IL2CPP/Gradle/unityLibrary/src/main/java/com/onesignal/UnityIAMLifecycleHandler.java:57: error: cannot find symbol
    public void onWillDismissInAppMessage(OSInAppMessage message) {
                                          ^
  symbol:   class OSInAppMessage
  location: class UnityIAMLifecycleHandler
/Users/jespermansa/Numfeud2023b/Library/Bee/Android/Prj/IL2CPP/Gradle/unityLibrary/src/main/java/com/onesignal/UnityIAMLifecycleHandler.java:63: error: cannot find symbol
    public void onDidDismissInAppMessage(OSInAppMessage message) {
                                         ^
  symbol:   class OSInAppMessage
  location: class UnityIAMLifecycleHandler
/Users/jespermansa/Numfeud2023b/Library/Bee/Android/Prj/IL2CPP/Gradle/unityLibrary/src/main/java/com/onesignal/UnityIAMLifecycleHandler.java:38: error: cannot find symbol
        void onWillDisplayInAppMessage(OSInAppMessage message);
                                       ^
  symbol:   class OSInAppMessage
  location: interface WrapperLifecycleHandler
/Users/jespermansa/Numfeud2023b/Library/Bee/Android/Prj/IL2CPP/Gradle/unityLibrary/src/main/java/com/onesignal/UnityIAMLifecycleHandler.java:39: error: cannot find symbol
        void onDidDisplayInAppMessage(OSInAppMessage message);
                                      ^
  symbol:   class OSInAppMessage
  location: interface WrapperLifecycleHandler
/Users/jespermansa/Numfeud2023b/Library/Bee/Android/Prj/IL2CPP/Gradle/unityLibrary/src/main/java/com/onesignal/UnityIAMLifecycleHandler.java:40: error: cannot find symbol
        void onWillDismissInAppMessage(OSInAppMessage message);
                                       ^
  symbol:   class OSInAppMessage
  location: interface WrapperLifecycleHandler
/Users/jespermansa/Numfeud2023b/Library/Bee/Android/Prj/IL2CPP/Gradle/unityLibrary/src/main/java/com/onesignal/UnityIAMLifecycleHandler.java:41: error: cannot find symbol
        void onDidDismissInAppMessage(OSInAppMessage message);
                                      ^
  symbol:   class OSInAppMessage
  location: interface WrapperLifecycleHandler
/Users/jespermansa/Numfeud2023b/Library/Bee/Android/Prj/IL2CPP/Gradle/unityLibrary/src/main/java/com/onesignal/UnityIAMLifecycleHandler.java:44: error: method does not override or implement a method from a supertype
    @Override
    ^
/Users/jespermansa/Numfeud2023b/Library/Bee/Android/Prj/IL2CPP/Gradle/unityLibrary/src/main/java/com/onesignal/UnityIAMLifecycleHandler.java:50: error: method does not override or implement a method from a supertype
    @Override
    ^
/Users/jespermansa/Numfeud2023b/Library/Bee/Android/Prj/IL2CPP/Gradle/unityLibrary/src/main/java/com/onesignal/UnityIAMLifecycleHandler.java:56: error: method does not override or implement a method from a supertype
    @Override
    ^
/Users/jespermansa/Numfeud2023b/Library/Bee/Android/Prj/IL2CPP/Gradle/unityLibrary/src/main/java/com/onesignal/UnityIAMLifecycleHandler.java:62: error: method does not override or implement a method from a supertype
    @Override
    ^
Note: /Users/jespermansa/Numfeud2023b/Library/Bee/Android/Prj/IL2CPP/Gradle/unityLibrary/src/main/java/com/unity3d/player/UnityPlayerActivity.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
13 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':unityLibrary:compileReleaseJavaWithJavac'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 10s
]
stdout[

> Configure project :launcher
WARNING: The option setting 'android.enableR8=false' is deprecated.
It will be removed in version 5.0 of the Android Gradle plugin.
You will no longer be able to disable R8
```


### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/OneSignalExample/Assets/OneSignal/CHANGELOG.md
+++ b/OneSignalExample/Assets/OneSignal/CHANGELOG.md
@@ -5,6 +5,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Fixed
+- Android - Lock OneSignal version so it doesn't get bumped to the next major version.
 ## [3.0.8]
 ### Changed
 - Renamed `enterLiveActivity` to `EnterLiveActivity` and `exitLiveActivity` to `ExitLiveActivity`

--- a/com.onesignal.unity.android/Editor/OneSignalAndroidDependencies.xml
+++ b/com.onesignal.unity.android/Editor/OneSignalAndroidDependencies.xml
@@ -3,6 +3,6 @@
     <repositories>
       <repository>https://repo.maven.apache.org/maven2</repository>
     </repositories>
-    <androidPackage spec="com.onesignal:OneSignal:4.8.3" />
+    <androidPackage spec="com.onesignal:OneSignal:[4.8.3]" />
   </androidPackages>
 </dependencies>

